boards/arm/sam4s_xplained/doc/sam4s_xplained.rst

```diff
@@ -0,0 +1,162 @@
+- IS66WV51216DALL 8 Mb SRAM
+- Micro-AB USB device
+- Micro-AB USB debug interface supporting SEGGER OB and Virtual COM Port and
+  Data
+- One reset and one user pushbutton
+- 2 yellow user LEDs
+- IC pads for external flash chip
+
+Supported Features
+==================
+
+The sam4s_xplained board configuration supports the following hardware
+features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | systick                             |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port                         |
++-----------+------------+-------------------------------------+
+| Ext SRAM  | on-chip    | external SRAM                       |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not currently supported by Zephyr.
+
+The default configuration can be found in the Kconfig
+:file:`boards/arm/sam4s_xplained/sam4s_xplained_defconfig`.
+
+Connections and IOs
+===================
+
+The `SAM4S Xplained Online User Guide`_ has detailed information about board
+connections. Download the `SAM4S Xplained Schematic`_ for more detail.
+
+System Clock
+============
+
+The SAM4S MCU is configured to use the 12 MHz internal oscillator on the board
+with the on-chip PLL to generate an 84 MHz system clock.
+
+Serial Port
+===========
+
+The ATSAM4S16C MCU has 2 UARTs and 2 USARTs. One of the UARTs (UART0) is
+connected to the Segger J-Link OB chip (the AT91SAM3U4 is programmed to be
+Segger J-Link OB). Segger J-Link OB brings the UART out as a virtual COM port.
+
+Programming and Debugging
+*************************
+
+The SAM4S Xplained board comes with Segger `J-Link OB <https://www.segger.com/jlink-ob.html>`_. This provides a debug
+interface to the SAM4S16C chip. You can use Ozone or JLink to communicate with
+the SAM4S16C.
+
+Flashing
+========
+
+#. Build the Zephyr kernel and the :ref:`hello_world` sample application:
+
+   .. code-block:: console
+
+      $ cd $ZEPHYR_BASE/samples/hello_world/
+      $ make BOARD=sam4s_xplained
+
+#. Connect the SAM4S Xplained board to your host computer using the USB debug
+   port.
+
+#. Run your favorite terminal program to listen for output. Under Linux the
+   terminal should be :code:`/dev/ttyACM0`. For example:
+
+   .. code-block:: console
+
+      $ minicom -D /dev/ttyACM0 -o
+
+   The -o option tells minicom not to send the modem initialization
+   string. Connection should be configured as follows:
+
+   - Speed: 115200
+   - Data: 8 bits
+   - Parity: None
+   - Stop bits: 1
+
+#. Download JLink from the Segger `JLink Downloads Page`_. Go to the section
```

Comment:
I should have commented earlier: in Zephyr we tend to use OpenOCD or other open source tools where possible rather than a proprietary ones. SAM4S Xplained board seems to be supported by OpenOCD so we probably should use it instead of JLink from Segger. In the future it would be also nice to have support for 'make flash' target. It's quite convenient and helps a lot for new users. That's said, so far no one else complained about this point and I'm also fine if this file is merged as it is at the moment.